### PR TITLE
Fixes bug where workflow approvals page would crash if deleted username was referenced

### PR DESCRIPTION
**Pull request by mabashian**
##### SUMMARY
link #9163 

The tooltip on the status labels for completed workflow approvals attempts to show the username of the user that acted on it.  If that user has been deleted then the key is not present in the object.  This would cause the page to crash.

To fix this, I conditionally show a different string when the username is not available.  If a user wants to see which deleted user acted on the approval they'll have to go to the activity stream.

Screen Shot 2021-04-22 at 9 59 23 AM

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 - UI

### CODE REVIEW

File: awx/ui_next/src/screens/WorkflowApproval/shared/WorkflowApprovalStatus.jsx [hunk 1]
```diff
@@ -28,11 +28,19 @@ function WorkflowApprovalStatus({ workflowApproval, i18n }) {
...
     return (
       <Tooltip
-        content={i18n._(
-          t`Denied by ${
-            workflowApproval.summary_fields.approved_or_denied_by.username
-          } - ${formatDateString(workflowApproval.finished)}`
-        )}
+        content={
+          workflowApproval.summary_fields?.approved_or_denied_by?.username
+            ? i18n._(
```
Comment (AlexSCorey) [inline awx/ui_next/src/screens/WorkflowApproval/shared/WorkflowApprovalStatus.jsx]: Remove the `i18n._` here in favor of  ```t`string` ```

If you can get rid of the `i18n` object and `withI18n` in this file as well that would be good to do.

File: awx/ui_next/src/screens/WorkflowApproval/shared/WorkflowApprovalStatus.jsx [hunk 2]
```diff
@@ -28,11 +28,19 @@ function WorkflowApprovalStatus({ workflowApproval, i18n }) {
...
+                t`Denied by ${
+                  workflowApproval.summary_fields.approved_or_denied_by.username
+                } - ${formatDateString(workflowApproval.finished)}`
+              )
+            : i18n._(
```
Comment (AlexSCorey) [inline awx/ui_next/src/screens/WorkflowApproval/shared/WorkflowApprovalStatus.jsx]: same

File: awx/ui_next/src/screens/WorkflowApproval/shared/WorkflowApprovalStatus.jsx [hunk 3]
```diff
@@ -45,11 +53,19 @@ function WorkflowApprovalStatus({ workflowApproval, i18n }) {
...
     return (
       <Tooltip
-        content={i18n._(
-          t`Approved by ${
-            workflowApproval.summary_fields.approved_or_denied_by.username
-          } - ${formatDateString(workflowApproval.finished)}`
-        )}
+        content={
+          workflowApproval.summary_fields?.approved_or_denied_by?.username
+            ? i18n._(
```
Comment (AlexSCorey) [inline awx/ui_next/src/screens/WorkflowApproval/shared/WorkflowApprovalStatus.jsx]: same

File: awx/ui_next/src/screens/WorkflowApproval/shared/WorkflowApprovalStatus.jsx [hunk 4]
```diff
@@ -45,11 +53,19 @@ function WorkflowApprovalStatus({ workflowApproval, i18n }) {
...
+                t`Approved by ${
+                  workflowApproval.summary_fields.approved_or_denied_by.username
+                } - ${formatDateString(workflowApproval.finished)}`
+              )
+            : i18n._(
```
Comment (AlexSCorey) [inline awx/ui_next/src/screens/WorkflowApproval/shared/WorkflowApprovalStatus.jsx]: same

**Review by AlexSCorey — CHANGES_REQUESTED** [review]
Looks good.  Small changes.

**Comment by mabashian** [comment]
@AlexSCorey I think I got all those cleaned up for ya

**Comment by tiagodread** [comment]
@mabashian could you rebase this one?

**Comment by tiagodread** [comment]
@mabashian could you rebase this one?

**Comment by one-t** [comment]
Just mentioning here, this PR may resolve some issues we're seeing with YOLO runs that include API tests. These seem to leave behind orphaned objects that cause the page to crash.

I'll see if I can figure out the exact reproducer(s). We want the UI to handle this but we also don't want orphaned objects floating around in the DB.

**Comment by one-t** [comment]
Ok, I've confirmed, the exact situation is this:
1) Create a workflow job template with an approval node
2) Create a user with the rights to approve
3) Launch workflow job and approve as the user
4) Delete the user
5) Page fails to render as described in the related issue.

I don't think any changes will be needed for the API tests, we already don't clean up jobs.

**Comment by one-t** [comment]
Confirmed, this does fix the issue and the page now renders when the approving user is deleted.

**Review by one-t — APPROVED** [review]
Jenkins run demonstrated that this fixes our issue with the API tests leaving behind orphaned approvals. :shipit:

**Comment by one-t** [comment]
!regate

**Review by AlexSCorey — APPROVED** [review]
Looks good well done!

**Pull request merged**